java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcFloat4VectorAccessorTest.java

```diff
@@ -193,9 +189,14 @@ public void testShouldGetBigDecimalMethodFromFloat4Vector() throws Exception {
         (accessor, currentRow) -> {
           float value = accessor.getFloat();
           if (Float.isInfinite(value) || Float.isNaN(value)) {
-            exceptionCollector.expect(SQLException.class);
+            assertThrows(
+                SQLException.class,
+                () -> {
+                  throw new SQLException();
+                });
+          } else {
+            assertThat(accessor.getBigDecimal(), is(BigDecimal.valueOf(value)));
```

Comment:
ETC. Is this change appropriate?

Reply:
I'm not sure this is right...

Reply:
I think it should be

```
if (isInfinite || isNaN) {
  assertThrows(SQLException.class, accessor::getBigDecimal);
} else {
  assertThat(accessor.getBigDecimal(), is(BigDecimal.valueOf(value));
}
```

Reply:
Fixed it.

https://github.com/apache/arrow/blob/323c70905aee03055d252924e8ab0a7b997d4f10/java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcFloat4VectorAccessorTest.java#L191-L195

https://github.com/apache/arrow/blob/323c70905aee03055d252924e8ab0a7b997d4f10/java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcFloat4VectorAccessorTest.java#L205-L211

https://github.com/apache/arrow/blob/323c70905aee03055d252924e8ab0a7b997d4f10/java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcFloat8VectorAccessorTest.java#L144-L148

https://github.com/apache/arrow/blob/323c70905aee03055d252924e8ab0a7b997d4f10/java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/accessor/impl/numeric/ArrowFlightJdbcFloat8VectorAccessorTest.java#L184-L190